### ISSUE TITLE
setting unique doesn't trigger an Exception : UniqueKeyViolation

### ISSUE THREAD

**Issue by sreecodeslayer** [josiahcarlson/rom#113]
Here is the model am after:

```
class SlaveServers(rom.Model):
	ip_address = rom.String(required=True, unique=True)
	free_slots = rom.Integer(required=True, default=10)
	status = rom.String(required=True, default='idle')
```
and I have a celery update on every 5 sec task that need to update this. 

```py
@celery.task
def update_redis():
	try:
		ss = SlaveServers(ip_address="192.168.10.22",status='idle')
		ss.save()
		print(SlaveServers.query.count()) # This keeps printing new count
	except Exception as e:
		# Update the existing document here with slots field [UniqueKeyViolation]
		raise e
```
The db with `ip_address` will be updated from diff slave servers if all works well, but this one seems to be an issue for now :thinking: 
Is there something am missing here to make it to work?

**Comment by josiahcarlson** [comment]
My conclusion: something is weird and broken. This shouldn't be happening, and I could have sworn I have tests for this.

**Comment by josiahcarlson** [comment]
Python 3.x+ seems to have an issue with saving data into unique indexes in some cases, not sure why.

**Comment by josiahcarlson** [comment]
In some cases, Python rom in 3.x would delete data from unique indexes after saving the data. This was caused by a weird "delete" case caused by rom not properly differentiating between missing column data on a new entity, and missing column data in a previously-saved entity. I'm pushing a fix now, and will be releasing version 0.40.0 to fix the issue.

Thank you for the bug report.

**Comment by josiahcarlson** [comment]
I'm closing as fixed. Thank you again for the fix, let me know if you run into any other issues.
